WcaOnRails/app/webpacker/components/Delegates/DelegatesOfRegion.jsx

```diff
@@ -77,38 +109,26 @@ function DelegatesTable({ delegates, isAdminMode }) {
 }
 
 export default function DelegatesOfRegion({ activeRegion, isAdminMode }) {
+  const isAllRegions = activeRegion.id === ALL_REGIONS.id;
   const { data: delegates, loading, error } = useLoadedData(
-    rolesOfGroup(activeRegion.id),
+    isAllRegions
+      ? apiV0Urls.userRoles.listOfGroupType(groupTypes.delegate_regions)
+      : rolesOfGroup(activeRegion.id),
   );
 
   const seniorDelegate = useMemo(
-    () => delegates?.find((delegate) => delegate.metadata.status === 'senior_delegate'),
-    [delegates],
+    () => (isAllRegions
+      ? null
+      : delegates?.find((delegate) => delegate.metadata.status === 'senior_delegate')),
```

Comment:
Does `!isAllRegions && delegates?.find(...)` work?

Reply:
But that will make it `false` instead of `null` right?

Reply:
Yes, you'd have to adjust the code later down to check for `=== false` instead of `=== null`

Reply:
Okay, I've done another approach now - using useCallback. Hope this is fine?